funsor/testing.py

```diff
@@ -72,7 +75,15 @@ def allclose(a, b, rtol=1e-05, atol=1e-08):
 
 def assert_close(actual, expected, atol=1e-6, rtol=1e-6):
     msg = ActualExpected(actual, expected)
-    assert type(actual) == type(expected), msg
+    if isinstance(actual, array):
+        assert isinstance(expected, array), msg
+    elif isinstance(actual, Tensor) and isinstance(actual.data, array):
+        assert isinstance(expected, Tensor) and isinstance(expected.data, array)
+    elif isinstance(actual, Contraction) and isinstance(actual.terms[0], Tensor) \
+            and isinstance(actual.terms[0].data, array):
+        assert isinstance(expected, Contraction) and isinstance(expected.terms[0].data, array)
+    else:
+        assert type(actual) == type(expected), msg
```

Comment:
nit: a newline after this final `else` clause would help separate the various `if-elif-sections`